src/client/index.js

```diff
@@ -43,12 +43,21 @@ export default function createClient(config, createStore) {
     })
   })
 
-  storage.getState()
+  Promise.all([storage.getState(), platformUtils.shouldRestartSettings()])
     .catch(function (err) {
       // eslint-disable-next-line no-console
       console.warn('Could not get persisted state', err)
       return []
```

Comment:
If the state cannot be retrieved for some reason, the `catch` return object shall match the next `then`:

```suggestion
      return [[], false]
```